src/System.Security.Permissions/src/System/Security/Permissions/PrincipalPermission.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace System.Security.Permissions
+{
+    public sealed partial class PrincipalPermission : System.Security.IPermission, System.Security.ISecurityEncodable, System.Security.Permissions.IUnrestrictedPermission
+    {
+        public PrincipalPermission(System.Security.Permissions.PermissionState state) { }
+        public PrincipalPermission(string name, string role) { }
+        public PrincipalPermission(string name, string role, bool isAuthenticated) { }
+        public System.Security.IPermission Copy() { return default(System.Security.IPermission); }
+        public void Demand() { }
```

Comment:
This is [https://github.com/dotnet/corefx/issues/9641](https://github.com/dotnet/corefx/issues/9641) ... @ioanacrant in that issue, I b elieve we decided we need this implemented to throw SecurityException, but PNSE is not a bad idea either

---
In reply to: [70821200](https://github.com/dotnet/corefx/pull/10049#discussion_r70821200) [](ancestors = 70821200)